src/ai/ai.cpp

```diff
@@ -166,8 +171,12 @@ void prune_alliances(sys::state& state) {
 		if(!n.get_is_player_controlled() && !n.get_ai_is_threatened() && !(n.get_overlord_as_subject().get_ruler())) {
 			prune_targets.clear();
 			for(auto dr : n.get_diplomatic_relation()) {
+				auto other = dr.get_related_nations(0) != n ? dr.get_related_nations(0) : dr.get_related_nations(1);
```

Comment:
I don't think there is any point putting this check outside of the are allied check. The prune list is only used to remove existing alliances, and checking for possible cbs against nations you aren't allied to here would be kind of a waste if there is nothing to prune.